### PR TITLE
Zhao

### PR DESCRIPTION
**Pull request by ThomasJuan23**
Finish("Register Page")

### CODE REVIEW

File: frontend/src/pages/Reg/Reg.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+          {
+            required: true,
+            message: 'Please choose your role!',
+          },
+        ]}
+      >
+        <Radio.Group
+          options={[
+            { label: "Customer", value: "customer"},
+            { label:"Provider", value: "provider"}
```
Comment (KingFarGrace) [inline frontend/src/pages/Reg/Reg.js:188]: value should be serviceProvider instead of provider

File: frontend/src/api/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export const reqRegister = (email, password,nickname,role) =>{
+    return ajax(
+        {
+            method: 'POST',
+            url: BASE + '/register',
+            data:{
+                email,
+                password,
+                nickname,
```
Comment (KingFarGrace) [inline frontend/src/api/index.js:27]: username instead of nickname

**Review by KingFarGrace — CHANGES_REQUESTED** [review]
- For a service provider, it can be accepted to login only when they get approved by admin. There will be a return value from backend which is called "available", true means that the admin approved the account's registration.
- Also, the service provider need to provide more information than a customer.
- 2 more comments need to be noticed in file /Reg/Reg.js and /api/index.js

**Comment by KingFarGrace** [comment]
为什么有两个pr，看起来是同一个分支

**Comment by ThomasJuan23** [comment]
> 为什么有两个pr，看起来是同一个分支
有一个错误的pull到master去了

**Comment by ThomasJuan23** [comment]
> * For a service provider, it can be accepted to login only when they get approved by admin. There will be a return value from backend which is called "available", true means that the admin approved the account's registration.
> * Also, the service provider need to provide more information than a customer.

1. 服务提供商需要提供的额外信息是我这边确定，还是你那边已经订好了
2. 前端需不需要做邮箱验证码的内容

**Comment by ThomasJuan23** [comment]
> * For a service provider, it can be accepted to login only when they get approved by admin. There will be a return value from backend which is called "available", true means that the admin approved the account's registration.

在服务提供商注册之后，如何让他们得知他们的注册通过了呢，可以通过邮件吗？我前段这边反馈给他们一个消息提醒叫他们等待确认邮件

**Pull request closed without merging**